### PR TITLE
Completed Web Components Notes

### PR DESCRIPTION
**Pull request by caydenakins**

### CODE REVIEW

File: web-components-f902/notes/00_javascript-render-hello-world-text-with-custom-elements.md.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# Render Hello World Text with Custom Elements
+
+[Video link](https://egghead.io/lessons/javascript-render-hello-world-text-with-custom-elements)
```
Comment (Creeland) [inline web-components-f902/notes/00_javascript-render-hello-world-text-with-custom-elements.md.md:3]: Video links aren't a requirement 🙂

File: web-components-f902/notes/00_javascript-render-hello-world-text-with-custom-elements.md.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+<TimeStamp start="0:09" end="0:17">
+
+We first define a class, `MyCustomElement()` which extends `HTMLElement` inside the `<script>` tags before creating our custom element.
+
+</TimeStamp>
+
+<TimeStamp start="0:29" end="0:37">
+
+We define a `connectedCallback` after defining our click event set up with a callback. A `connectedCallback` is invoked each time the custom element is appended into a document-connected element.
```
Comment (Creeland) [inline web-components-f902/notes/00_javascript-render-hello-world-text-with-custom-elements.md.md:13]: I'd shorten this to just "A `connectedCallback` is invoked each time the custom element is appended into a document-connected element."

File: web-components-f902/notes/00_javascript-render-hello-world-text-with-custom-elements.md.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+<TimeStamp start="0:45" end="0:54">
+
+To define our custom element:
+
+```jsx
+window.customElements.define('my-custom-element', MyCustomElement);
+```
+
+Custom element tag names must be lowercase, contain a dash (helps distinguish from regular elements), and cannot be self-closing. Additionally, the same tag cannot be registered more than once.
```
Comment (Creeland) [inline web-components-f902/notes/00_javascript-render-hello-world-text-with-custom-elements.md.md:25]: Remove the lines above this sentence. Reduce sentence to following
```suggestion
Custom element tag names must be lowercase, contain a dash (helps distinguish from regular elements), cannot be self-closing, and cannot be registered more than once.
```

File: web-components-f902/notes/01_html-5-build-flexible-html-with-htmltemplates-using-slots-and-web-components.md.md [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+
+<TimeStamp start="2:43" end="2:48">
+
+Templates can also be used dynamically in conjunction with dynamic content.
+
+</TimeStamp>
+
+<TimeStamp start="2:50" end="3:02">
+
+We can accomplish this by copying the template's content into a string (`templateString`), changing the template definition to `const template = document.createElement('template');`, and setting `template.innerHTML = templateString;`.
```
Comment (Creeland) [inline web-components-f902/notes/01_html-5-build-flexible-html-with-htmltemplates-using-slots-and-web-components.md.md:91]: Try to not recreate the instructions in the video 1 to 1. Ask yourself how you're adding to the instruction or how you can summarize it succinctly.

File: web-components-f902/notes/01_html-5-build-flexible-html-with-htmltemplates-using-slots-and-web-components.md.md [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+</TimeStamp>
+
+<TimeStamp start="2:26" end"2:36">
+
+After reusing our custom element, we're able to observe how powerful `<slot>` can be. Dynamic content can be achieved with this code:
+
+<custom-element>
+    <iframe slot="slot1" width="560" height="315" src="https://www.youtube.com/embed/Bv_5Zv5c-Ts" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+    <div slot="slot2">Slot 2 text inside a div</div>
+</custom-element>
```
Comment (Creeland) [inline web-components-f902/notes/01_html-5-build-flexible-html-with-htmltemplates-using-slots-and-web-components.md.md:79]: This is a lot of code for a note. A good rule of thumb I've found is to keep notes to about the size of a tweet.

Comment (Creeland) [reply web-components-f902/notes/01_html-5-build-flexible-html-with-htmltemplates-using-slots-and-web-components.md.md:79]: Maybe instead of saying "this code:" followed by the code block, you could share a small snippet of the relevant part of the iframe. like `<iframe ... {the relevant part} ...></iframe>`

Also remember, if you use a code block to put a code fence ``` above and below the block

File: web-components-f902/notes/03_html-5-allow-external-styling-of-a-web-component-s-shadow-dom.md.md [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+
+<TimeStamp start="1:51" end="2:00">
+
+In the console: `$0.changeStyles({color: 'red', textDecoration: 'underline'})`
+
+</TimeStamp>
+
+<TimeStamp start="2:24" end="2:34">
+
+`attributeChangedCallback()` works in conjunction with our getter method, `observedAttributes()`, by firing anytime a custom element's attributes is added, removed, or changed. Specifically, we are looking for changes in our `'color'` and `'text-decoration'` attributes.
```
Comment (Creeland) [inline web-components-f902/notes/03_html-5-allow-external-styling-of-a-web-component-s-shadow-dom.md.md:31]: rework this note. 
```suggestion
`attributeChangedCallback()` works in conjunction with our getter method, `observedAttributes()`, by firing anytime a custom element's attributes is modified
```

File: web-components-f902/notes/03_html-5-allow-external-styling-of-a-web-component-s-shadow-dom.md.md [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+
+</TimeStamp>
+
+<TimeStamp start="3:12" end="3:19">
+
+In the console: `$0.setAttribute('color', 'purple')`
+
+</TimeStamp>
+
+<TimeStamp start="4:35" end="4:49">
```
Comment (Creeland) [inline web-components-f902/notes/03_html-5-allow-external-styling-of-a-web-component-s-shadow-dom.md.md:41]: Remove the recap

File: web-components-f902/notes/04_javascript-web-components-explicit-api-s-using-class-methods.md.md [hunk 8]
```diff
@@ -0,0 +1,41 @@
+# Web Components explicit API's using class methods
+
+[Video link](https://egghead.io/lessons/javascript-web-components-explicit-api-s-using-class-methods)
+
+<TimeStamp start="0:18" end="0:28">
+
+```jsx
```
Comment (Creeland) [inline web-components-f902/notes/04_javascript-web-components-explicit-api-s-using-class-methods.md.md:7]: Are you using the code block notes so the user can conveniently copy?

**Review by Creeland — CHANGES_REQUESTED** [review]

File: web-components-f902/notes/04_javascript-web-components-explicit-api-s-using-class-methods.md.md (hunk 8 above)
Comment (caydenakins) [reply web-components-f902/notes/04_javascript-web-components-explicit-api-s-using-class-methods.md.md:7]: Sort of, this was establishing the baseline at the start of the lesson and was trying to find a way to help contribute to a viewer's experience. I realize now that if they're looking to copy code, they would have the actual lesson code pulled up, so this is likely worth removing.

File: web-components-f902/notes/01_html-5-build-flexible-html-with-htmltemplates-using-slots-and-web-components.md.md (hunk 5 above)
Comment (caydenakins) [reply web-components-f902/notes/01_html-5-build-flexible-html-with-htmltemplates-using-slots-and-web-components.md.md:79]: Completely missed the code fence on this one lol. Thanks for the catch! And definitely a lot of code, I'll reword that.
